src/OpenSleigh.Transport.RabbitMQ/QueueReferenceFactory.cs

```diff
@@ -1,10 +1,13 @@
 ﻿using System;
+using System.Collections.Concurrent;
 using OpenSleigh.Core.Messaging;
 
 namespace OpenSleigh.Transport.RabbitMQ
 {
-    public class QueueReferenceFactory : IQueueReferenceFactory
+    internal class QueueReferenceFactory : IQueueReferenceFactory
```

Comment:
Personally I don't like this kind of encapsulation anymore, nowadays I really prefer the way adopted that many frameworks to have all the interfaces and classes public, but put the "non-public" one inside a namespace `Internal` this leave the ability to have access to the internals at your own risk. The read public surface follow the semantic versioning, the internal surface can be broken at the owner's discretion.

On of the example is [entity framework core](https://github.com/dotnet/efcore/tree/main/src/EFCore/Internal).

In every internal class would be nice to add a similar disclaimer:

```csharp
    /// <summary>
    ///     This is an internal API that supports the OpenSleigh infrastructure and not subject to
    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
    ///     any release. You should only use it directly in your code with extreme caution and knowing that
    ///     doing so can result in application failures when updating to a new OpenSleigh release.
    /// </summary>
```

Reply:
yeah you're right. I just recently had a bug due to this excessive encapsulation. I'll get rid of that.